sdk/python/tests/expediagroup/test_milvus_online_store.py

```diff
@@ -1,37 +1,346 @@
     online_store_creator.create_online_store()
 
-    # access through a stub
-    milvus = Milvus(online_store_creator.host, 19530)
-
-    # Create collection demo_collection if it doesn't exist.
-    collection_name = "example_collection_"
-
-    ok = milvus.has_collection(collection_name)
-    if not ok:
-        fields = {
-            "fields": [
-                {"name": "key", "type": DataType.INT64, "is_primary": True},
-                {
-                    "name": "vector",
-                    "type": DataType.FLOAT_VECTOR,
-                    "params": {"dim": 32},
+    yield online_store_creator
+
+    # Tearing down the Milvus instance after all tests in the class
+    online_store_creator.teardown()
+
+
+class PymilvusConnectionContext:
+    def __enter__(self):
+        # Connecting to Milvus
+        connections.connect(host=HOST, port=PORT)
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        # Disconnecting from Milvus
+        connections.disconnect("milvus")
+
+
+class TestMilvusConnectionManager:
+    def test_connection_manager(self, repo_config, caplog, milvus_online_setup, mocker):
+
+        mocker.patch("pymilvus.connections.connect")
+        with MilvusConnectionManager(repo_config.online_store):
+            assert (
+                f"Connecting to Milvus with alias {repo_config.online_store.alias} and host {repo_config.online_store.host} and default port {repo_config.online_store.port}."
+                in caplog.text
+            )
+
+        connections.connect.assert_called_once_with(
+            host=repo_config.online_store.host,
+            username=repo_config.online_store.username,
+            password=repo_config.online_store.password,
+            use_secure=True,
+        )
+
+    def test_context_manager_exit(
+        self, repo_config, caplog, milvus_online_setup, mocker
+    ):
+        # Create a mock for connections.disconnect
+        mock_disconnect = mocker.patch("pymilvus.connections.disconnect")
+
+        # Create a mock logger to capture log calls
+        mock_logger = mocker.patch(
+            "feast.expediagroup.vectordb.milvus_online_store.logger", autospec=True
+        )
+
+        with MilvusConnectionManager(repo_config.online_store):
+            print("Doing something")
+
+        # Assert that connections.disconnect was called with the expected argument
+        mock_disconnect.assert_called_once_with(repo_config.online_store.alias)
+
+        with pytest.raises(Exception):
+            with MilvusConnectionManager(repo_config.online_store):
+                raise Exception("Test Exception")
+        mock_logger.error.assert_called_once()
+
+
+class TestMilvusOnlineStore:
+
+    collection_to_write = "Collection2"
+    collection_to_delete = "Collection1"
+
+    def setup_method(self, milvus_online_setup):
+        # Ensuring that the collections created are dropped before the tests are run
+        with PymilvusConnectionContext():
```

Comment:
it's ok to re-use `MilvusConnectionManager` in order to not duplicate code

Reply:
I want to keep the test not dependent on any functionality in Milvus online store. This context manager is a very simple open and close connection, purely for testing purposes. Will not add in username, password, etc.